### ISSUE TITLE
Detect redis connection failure

### ISSUE THREAD

**Issue by suryapandian** [badboy/redlock-rs#13]
Right now when we try to acquire lock, we get NONE when
- redis connection fails
- the lock is already acquired

We are not able to differentiate between the two.

We should expose a function that gives redis connectivity so that we can be sure the lock failure is because the lock is already acquired.

**Comment by suryapandian** [comment]
@aig787 

with 
```rs
        lock.servers[0].get_connection();
        if conn.is_err() {
            return Err(anyhow!(
                "Error acquiring lock! Please check redis connection"
            ));
        }
```

I handled this in my application. Since the return object is an `Option` and not a `Result`, I am not sure how we could handle this in the library without a breaking change.

**Comment by aig787** [comment]
Yep, it'll require a breaking change but I do think the change is a good one. Give #14 a shot, it should meet your needs.

**Comment by suryapandian** [comment]
> Yep, it'll require a breaking change but I do think the change is a good one. Give #14 a shot, it should meet your needs.
thanks will go through it sometime this week 👀

**Comment by aig787** [comment]
Resolved with 2.0.0

**Comment by suryapandian** [comment]
> Resolved with 2.0.0

thanks :)
